src/proxy/server.rs

```diff
@@ -99,14 +99,19 @@ struct ProcessDownstreamReceiveConfig {
 impl Server {
     /// start the async processing of incoming UDP packets. Will block until an
     /// event is sent through the stop Receiver.
-    pub async fn run(self, mut shutdown_rx: watch::Receiver<()>) -> Result<()> {
+    pub async fn run(
+        self,
+        ready_tx: oneshot::Sender<SocketAddr>,
```

Comment:
Well in my attempts to remove this, and replace it with polling, I start running into an issue where the server shuts down before the tests can complete, and I can't trace where it is causing it to shutdown. So for now, I'm going to move to try and simplify our task management, so that's there's a much clearer and easier to track how we're shutting down.

Reply:
Maybe a silly question - but what happens if you don't have a the `ready_tx` at all? 

Out integration tests in Rust don't seem to need to wait for the ready state, so I'm curious as to why this does?